File: app/src/main/java/net/north101/android/ghplayertracker/data/SelectableCharacter.java
package net.north101.android.ghplayertracker.data;

import org.parceler.Parcel;
import org.parceler.ParcelConstructor;

@Parcel
public class SelectableCharacter {
    public final Character character;
    public boolean selected;

    @ParcelConstructor
    public SelectableCharacter(Character character, boolean selected) {
        this.character = character;
        this.selected = selected;
    }
}

File: app/src/main/java/net/north101/android/ghplayertracker/data/SelectableCharacter.kt
package net.north101.android.ghplayertracker.data

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import net.north101.android.ghplayertracker.RecyclerItemCompare

@Parcelize
data class SelectableCharacter(
    val character: Character,
    var selected: Boolean
) : Parcelable, RecyclerItemCompare {
    override val compareItemId: String
        get() = character.id.toString()

    fun copy(): SelectableCharacter {
        return SelectableCharacter(character.copy(), selected)
    }
}
